ambry-clustermap/src/main/java/com.github.ambry.clustermap/StaticClusterAgentsFactory.java

```diff
@@ -75,9 +76,15 @@ public StaticClusterManager getClusterMap() {
   public ClusterParticipant getClusterParticipant() throws IOException {
     if (clusterParticipant == null) {
       clusterParticipant = new ClusterParticipant() {
+        private final List<PartitionStateChangeListener> listeners = new LinkedList<>();
+
         @Override
         public void participate(List<AmbryHealthReport> ambryHealthReports) {
-
+          for (PartitionStateChangeListener listener : listeners) {
+            for (String partitionName : partitionLayout.getAllPartitionNames()) {
+              listener.onPartitionLeadFromStandby(partitionName);
```

Comment:
Do we know that it was a standby before participate was called?

Reply:
This is the case for static cluster map. There will be no event for either standby from leader, or leader from standby in this case. The logic above just forces callbacks simulating adding partition leader. For this we don't care if partition was a standby or not.